components/chainhook-sdk/src/observer/mod.rs

```diff
@@ -418,6 +433,9 @@ pub async fn start_event_observer(
 
     let background_job_tx_mutex = Arc::new(Mutex::new(observer_commands_tx.clone()));
 
+    let observer_metrics = ObserverMetrics::default();
+    let observer_metrics_arc = Arc::new(Mutex::new(observer_metrics));
```

Comment:
Could we use a `RwLock`, instead of a `Mutex`?
The API is always reading, and the runloop writing.